flink-table/flink-sql-parser/src/main/codegen/data/Parser.tdd

```diff
@@ -147,11 +149,14 @@
     "COMPILE"
     "COLUMNS"
     "DATABASES"
+    "DISTRIBUTED"
+    "DISTRIBUTION"
     "DRAIN"
     "ENFORCED"
     "ESTIMATED_COST"
     "EXTENDED"
     "FUNCTIONS"
+    "HASH"
```

Comment:
Ok, I tried removing HASH as a reserved keyword.

I am hitting issues around RANGE being a keyword.  I'm trying to read the distributionKind using `SqlIdentifier` and I haven't quite figured out how to make that work.

The FLIP didn't have the algorithm as quoted, so that's why I added HASH as a reserved keyword.  Are you suggesting that algorithm (if supplied) must be quoted?

Reply:
I wasn't aware that RANGE is already a reserved keyword. This of course changes the picture, we don't want to force users using backticks by default. If this is the case, I guess we have to make HASH a reserved keyword as well.

Reply:
Do you know the difference between `nonReservedKeywords`, `reservedKeywords`, and `keywords`? I'm wondering if it at least makes sense to still allow `hash` as a column name without backticks if we add it to the right list. We currently have too many keywords which basically forces the user to use backticks all the time.